app/Http/Controllers/Auth/LoginController.php

```diff
@@ -95,23 +95,16 @@ protected function setVatsimAuth($userId)
 
     protected function attemptSecondaryAuth()
     {
-        if (Session::has('url.intended')) {
-            $intended = Session::get('url.intended');
-            if (starts_with($intended, route('password.request'))) {
-                Session::remove('url.intended');
-
-                return redirect($intended);
-            }
-        }
-
+        $intended = Session::pull('url.intended', route('site.home'));
```

Comment:
This needs to be after line 102, otherwise the next time they come around after secondary pass, the intended destination will already have been removed from the session.

Reply:
I'm not quite sure you understood what I meant here - by having the `$intended` where it is in the code, a user who has to attempt secondary authentication (i.e gets taken to put in their secondary password), will not be redirected to the intended page. I was simply saying to move the pull of the intended url to below the `hasPassword` check so that this doesn't happen